### PR TITLE
Improve comments / abstractions

### PR DESCRIPTION
**Pull request by ValarDragon**
## What is the purpose of the change

Aim to improve comments / logic in main function 

## Brief Changelog

*(for example:)*
 
  - *The metadata is stored in the blob store on job creation time as a persistent artifact*
  - *Deployments RPC transmits only the blob storage reference*
  - *Daemons retrieve the RPC data from the blob cache*

## Testing and Verifying

*(Please pick one of the following options)*

This change is a trivial rework / code cleanup without any test coverage.

*(or)*

This change is already covered by existing tests, such as *(please describe tests)*.

*(or)*

This change added tests and can be verified as follows:

*(example:)*
  - *Added unit test that validates ...*
  - *Added integration tests for end-to-end deployment with ...*
  - *Extended integration test for ...*
  - *Manually verified the change by ...*

## Documentation and Release Note

  - Does this pull request introduce a new feature or user-facing behavior changes? (yes / no)
  - Is a relevant changelog entry added to the `Unreleased` section in `CHANGELOG.md`? (yes / no)
  - How is the feature or change documented? (not applicable   /   specification (`x/<module>/spec/`)  /  [Osmosis docs repo](https://github.com/osmosis-labs/docs)   /   not documented)

### CODE REVIEW

File: x/gamm/pool-models/balancer/amm.go [hunk 1]
```diff
@@ -271,32 +271,46 @@ func (p *Pool) JoinPool(_ctx sdk.Context, tokensIn sdk.Coins, swapFee sdk.Dec) (
...
+// The input tokens must either be:
+// - a single token
+// - contain exactly the same tokens as the pool contains
+//
+// It returns the number of shares created, the amount of coins actually joined into the pool
+// (in case of not being able to fully join), or an error.
+func (p *Pool) CalcJoinPoolShares(_ sdk.Context, tokensIn sdk.Coins, swapFee sdk.Dec) (numShares sdk.Int, tokensJoined sdk.Coins, err error) {
+	// 1) Get pool current liquidity + and token weights
+	// 2) If single token provided, do single asset join and exit.
+	// 3) If multi-asset join, first do as much of a join as we can with no swaps.
```
Comment (stackman27) [inline x/gamm/pool-models/balancer/amm.go:283]: Are we mention about when swapFee is used during multi-asset join?

Comment (ValarDragon) [reply x/gamm/pool-models/balancer/amm.go:283]: added a one line note after the 5 steps.

**Review by p0mvn — APPROVED** [review]
LGTM

File: x/gamm/pool-models/balancer/amm.go [hunk 2]
```diff
@@ -306,32 +321,35 @@ func (p *Pool) CalcJoinPoolShares(_ sdk.Context, tokensIn sdk.Coins, swapFee sdk
...
 	// for each of them.
-	if !remainingTokensIn.Empty() {
-		// update pool assets with newLiquidity for accurate calcSingleAssetJoin calculation.
-		if err := updateIntermediaryPoolAssetsLiquidity(newLiquidity, poolAssetsByDenom); err != nil {
-			return sdk.ZeroInt(), sdk.NewCoins(), err
-		}
-
-		// update total shares for accurate calcSingleAssetJoin calculation.
-		newTotalShares := totalShares.Add(numShares)
+	// so first we update the pool state for this.
```
Comment (alexanderbez) [inline x/gamm/pool-models/balancer/amm.go:333]:
```suggestion
	// So first we update the pool state for this, then we
```

**Review by alexanderbez — APPROVED** [review]

**Comment by ValarDragon** [comment]
Will merge after tests finish

**Pull request merged**